src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Bucket<K, V>[] table;
+    private int size = 0;
+
+    public MyHashMap() {
+        table = new Bucket[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        checkCapacity();
+        int position = getPosition(key);
+        Bucket<K, V> bucket = table[position];
+        Bucket<K, V> newBucket = new Bucket<>(key, value);
+        if (table[position] == null) {
+            table[position] = newBucket;
+        } else {
+            if (Objects.equals(bucket.key, newBucket.key)) {
+                bucket.value = newBucket.value;
+                return;
+            }
+            Bucket<K, V> tempBucket = bucket;
+            while (tempBucket.next != null) {
+                tempBucket = tempBucket.next;
+                if (Objects.equals(tempBucket.key, newBucket.key)) {
+                    tempBucket.value = newBucket.value;
+                    return;
+                }
+            }
+            tempBucket.next = newBucket;
+        }
+        size++;
+    }
 
+    private int getPosition(K key) {
+        int position;
+        if (key == null) {
+            position = 0;
+        } else {
+            position = Math.abs(key.hashCode() % table.length);
+        }
+        return position;
+    }
```

Comment:
always use `Math.abs(key.hashCode() % table.length)`, read about KISS principle, no sense to make it harder than it is